File: emcrypt-core/src/main/java/com/beam/emcryptcore/model/auth/AuthResponse.java
package com.beam.emcryptcore.model.auth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
@Data
@Builder
public class AuthResponse {

    private int code;
    private String token;
    private Account account;
}

File: emcrypt-web/src/main/ui/src/app/modules/auth/models/auth-response.ts
import { Account } from './account';

export class AuthResponse {
  code: number;
  token: string;
  account: Account;

  setAuth(auth: AuthResponse) {
    this.code = auth.code;
    this.token = auth.token;
    this.account = auth.account;
  }
}
